streams/src/main/java/org/apache/kafka/streams/kstream/internals/PrintForeachAction.java

```diff
@@ -51,6 +51,9 @@
     public void apply(final K key, final V value) {
         final String data = String.format("[%s]: %s", label, mapper.apply(key, value));
         printWriter.println(data);
+        if (!closable) {
```

Comment:
So then this won't work as intended for `System.out` and `System.err`...

Reply:
Can you elaborate?

Reply:
Sorry.  You are right.

Reply:
No need to be sorry. Glad we could clarify, and that the code is correct and we did not miss something :)